fiat/coincap_api.go

```diff
@@ -37,94 +30,94 @@ var (
 
 // Granularity indicates the level of aggregation price information will be
 // provided at.
-type Granularity string
+type Granularity time.Duration
 
-const (
+var (
 	// GranularityMinute aggregates the bitcoin price over 1 minute.
-	GranularityMinute Granularity = "m1"
+	GranularityMinute = Granularity(time.Minute)
 
-	// GranularityMinute aggregates the bitcoin price over 1 minute.
-	Granularity5Minute Granularity = "m5"
+	// Granularity5Minute aggregates the bitcoin price over 5 minute.
+	Granularity5Minute = Granularity(time.Minute * 5)
 
-	// GranularityMinute aggregates the bitcoin price over 15 minutes.
-	Granularity15Minute Granularity = "m15"
+	// Granularity15Minute aggregates the bitcoin price over 15 minutes.
+	Granularity15Minute = Granularity(time.Minute * 15)
 
-	// GranularityMinute aggregates the bitcoin price over 30 minutes.
-	Granularity30Minute Granularity = "m30"
+	// Granularity30Minute aggregates the bitcoin price over 30 minutes.
+	Granularity30Minute = Granularity(time.Minute * 30)
 
 	// GranularityHour aggregates the bitcoin price over 1 hour.
-	GranularityHour Granularity = "h1"
+	GranularityHour = Granularity(time.Hour)
 
 	// Granularity6Hour aggregates the bitcoin price over 6 hours.
-	Granularity6Hour Granularity = "h6"
+	Granularity6Hour = Granularity(time.Hour * 6)
 
 	// Granularity12Hour aggregates the bitcoin price over 12h hours.
-	Granularity12Hour Granularity = "h12"
+	Granularity12Hour = Granularity(time.Hour * 12)
 
 	// GranularityDay aggregates the bitcoin price over one day.
-	GranularityDay Granularity = "d1"
+	GranularityDay = Granularity(time.Hour * 24)
 )
 
-// maxDuration returns the maximum duration that can be queried for a given
-// level of granularity.
-func (g Granularity) maxDuration() (time.Duration, error) {
+// coincapValue returns the value we need to provide the API with for each level
+// of granularity.
+func (g Granularity) coincapValue() (string, error) {
```

Comment:
If `Granularity` was just a struct with `{Duration time.Duration, CoincapValue string, MaxDuration time.Duration}` these methods could be removed. Would also remove the need for an error return value.
And all values could be in the same place, for example:
`GranularityMinute = newGranularity(time.Minute, "m1", time.Hour * 24)`

Reply:
Indeed, much nicer this way :)